QuadradiusR/scripts/network_handler.gd

```diff
@@ -40,16 +46,30 @@ func _create_user_1(message: Message, args: Dictionary):
 	pass
 
 
+# # # # # # # # # #
+# AUTHORIZE USER  #
+# # # # # # # # # #
 func authorize_user(username: String, password: String, cb: FuncRef, args: Dictionary):
 	rest_api.authorize(username, password, funcref(self, "_authorize_user_1"), {'cb': cb, 'user_data': args})
 
 func _authorize_user_1(message: Message, args: Dictionary):
 	var is_authorized = message.result == 200
-	username = args['user_data']['username']  # TODO only if successful
-	token = message.body['token'] if is_authorized else ""
-	args['cb'].call_func(is_authorized, args['user_data'], message)
+
+	if is_authorized:
+		token = message.body['token'] if is_authorized else ""
+		rest_api.get_user_me(self.token, funcref(self, "_authorize_user_2"), args)
+	else:
+		args['cb'].call_func(is_authorized, args['user_data'], message)
+
+func _authorize_user_2(message: Message, args: Dictionary):
+	username = message.body['username']
+	user_id = message.body['id']
```

Comment:
`self.username`, `self.user_id` or maybe put in a new `Context` class

Reply:
Moved to new class